docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,819 @@
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0,
+    0, // one more dummy byte for the SigHashType
+	}
+  // The compressed public key requires 33 bytes.
+  dummyCompressedPublicKeyForWitness := []byte{
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0,
+  }
+
+	tx := wire.NewMsgTx(wire.TxVersion)
+	txIn := wire.NewTxIn(
+		wire.NewOutPoint(previousOutputTransactionHash, previousOutputIndex),
+		nil, // scriptSig is empty here
+		[][]byte{
+      dummySignatureForWitness,
+      dummyCompressedPublicKeyForWitness,
+    },
+	)
+	txIn.Sequence = 0
+	tx.AddTxIn(txIn)
+
+	for _, recipientAddress := range recipientAddresses {
+		address, err := btcutil.DecodeAddress(recipientAddress, chainParams)
+		if err != nil {
+			return "", fmt.Errorf(
+				"error decoding output address [%s]: [%s]",
+				recipientAddress,
+				err,
+			)
+		}
+		outputScript, err := txscript.PayToAddrScript(address)
+		if err != nil {
+			return "", fmt.Errorf(
+				"error constructing script from output address [%s]: [%s]",
+				recipientAddress,
+				err,
+			)
+		}
+
+		tx.AddTxOut(wire.NewTxOut(
+			int64(0), // value is filled in after fee is computed below
+			outputScript,
+		))
+	}
+
+	// Compute weight and vsize per [BIP141], except vsize is truncated
+	// instead of rounded up, then compute the final fee and set the
+	// per-recipient value. Could result in a fractionally low fee.
+	weight := tx.SerializeSizeStripped()*3 + tx.SerializeSize()
+	vsize := weight / 4
+	fee := feePerVbyte * int64(vsize)
+	perRecipientValue := (previousOutputValue - fee) / int64(len(recipientAddresses))
+	for _, txOut := range tx.TxOut {
+		txOut.Value = perRecipientValue
+	}
+
+	return tx, nil
+}
+```
+<1> These two come from `fundingInfo().utxoOutpoint`, which has the transaction
+    id as the first 32 little-endian bytes and the previous output index as
+    its last 4 little-endian bytes.
+<2> Comes from `fundingInfo().utxoValue`, a little-endian unsigned int.
+<3> Derived as part of the <<signing-protocol>>.
+<4> e.g. &chaincfg.MainNetParams or TestNet3Params
+
+
+[[sighash-construction]]
+==== Construction of Sighash for Signing
+
+The sighash is constructed as the double-SHA256 of the preimage. The preimage
+is defined in BIP143. Fortunately, once the dummy unsigned transaction is
+available, btcutil provides the tooling to build the sighash. The sample
+function call below shows how to construct the sighash once an unsigned
+transaction is available from above the function in
+<<transaction-construction>>. This sighash is the data then signed by the
+signer group to produce a valid transaction (see the section on
+<<final-transaction-construction>> for the rest of the process once the
+signature has been computed):
+
+```go
+// import "github.com/btcsuite/btcd/txscript"
+
+// unsignedTransaction := ConstructUnsignedTransaction(...)
```

Comment:
Out of curiosity, why are these line and import lines prepended with `//`?

Reply:
For imports, they're not designed to be standalone files. For this line, it's not described here and would be run elsewhere based on various inputs.